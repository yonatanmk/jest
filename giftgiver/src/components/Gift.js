import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';

class Gift extends Component {
  constructor(props) {
    super(props)

    this.state = {
      person: '',
      present: ''
    }
  }


  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <ControlLabel>Person</ControlLabel>
            <FormControl
              id='input-person'
              onChange={event => this.setState({ person: event.target.value })}
              />
          </FormGroup>
          <FormGroup>
            <ControlLabel>Present</ControlLabel>
            <FormControl
              id='input-present'
              onChange={event => this.setState({ present: event.target.value })}
              />
          </FormGroup>
        </Form>
        <Button id='btn-remove' onClick={() => this.props.removeGift(this.props.gift.id)} >Remove Gift</Button>
      </div>
    )
  }
}

export default Gift;
