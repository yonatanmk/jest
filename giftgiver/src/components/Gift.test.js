import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift'

describe('Gift', () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = {
    gift: { id },
    removeGift: mockRemove,
  }
  const gift = shallow(<Gift {...props} />);

  it('renders correctly', () => {
    expect(gift).toMatchSnapshot();
  });

  it('initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({ person: '', present: '' });
    expect(gift.state().person).toEqual('');
    expect(gift.state().present).toEqual('');
  });

  describe('When typing into the person input', () => {
    const person = 'Uncle';

    beforeEach(() => {
      gift.find('#input-person').simulate('change', { target: { value: person } }); // 2nd arg is simulated event object, sec 2 lec 14 7:00
    })

    it('updates the person in `state`', () => {
      expect(gift.state().person).toEqual(person);
    });
  })

  describe('When typing into the present input', () => {
    const present = 'Golf Clubs';

    beforeEach(() => {
      gift.find('#input-present').simulate('change', { target: { value: present } }); // 2nd arg is simulated event object, sec 2 lec 14 7:00
    })

    it('updates the present in `state`', () => {
      expect(gift.state().present).toEqual(present);
    });
  })

  describe('When clicking the `Remove Gift` button', () => {
    beforeEach(() => {
      gift.find('#btn-remove').simulate('click');
    })

    it('calls the removeGift callback', () => {
      expect(mockRemove).toHaveBeenCalled();
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  })
})
