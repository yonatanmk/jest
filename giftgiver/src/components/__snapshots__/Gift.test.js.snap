// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
      Object {
        "id": 1,
      }
    }
    removeGift={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Person
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              id="input-person"
              onChange={[Function]}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Present
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              id="input-present"
              onChange={[Function]}
            />
          </FormGroup>
        </Form>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          id="btn-remove"
          onClick={[Function]}
        >
          Remove Gift
        </Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Person
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                id="input-person"
                onChange={[Function]}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Present
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                id="input-present"
                onChange={[Function]}
              />
            </FormGroup>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Person
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="input-person"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Person",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Person",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "id": "input-person",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Present
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="input-present"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Present",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Present",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "id": "input-present",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Remove Gift",
          "disabled": false,
          "id": "btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Gift",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Person
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                id="input-person"
                onChange={[Function]}
              />
            </FormGroup>
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Present
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                id="input-present"
                onChange={[Function]}
              />
            </FormGroup>
          </Form>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            id="btn-remove"
            onClick={[Function]}
          >
            Remove Gift
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Person
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="input-person"
                  onChange={[Function]}
                />
              </FormGroup>,
              <FormGroup
                bsClass="form-group"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Present
                </ControlLabel>
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="input-present"
                  onChange={[Function]}
                />
              </FormGroup>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Person
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    id="input-person"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Person",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Person",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "id": "input-person",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
                  >
                    Present
                  </ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    id="input-present"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Present",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Present",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "id": "input-present",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Remove Gift",
            "disabled": false,
            "id": "btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Gift",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
